### PR TITLE
Gyro Inversion

### PR DESCRIPTION
**Pull request by jeffaburke**

### CODE REVIEW

File: core/src/main/java/com/chopshop166/chopshoplib/sensors/PigeonGyro.java [hunk 1]
```diff
@@ -46,27 +48,47 @@ public void close() throws Exception {
...
     public void calibrate() {
         // NoOp
     }
 
+    /**
+	 * Inverts the angle and rate of the Pigeon
+	 *
+	 * @param isInverted The state of inversion, true is inverted.
+	 */
+    public void setInverted(final boolean isInverted) {
```
Comment (msoucy) [inline core/src/main/java/com/chopshop166/chopshoplib/sensors/PigeonGyro.java:91]: Move nonderived functions to the top of the class please. Also fix the tabs.

File: core/src/main/java/com/chopshop166/chopshoplib/sensors/PigeonGyro.java [hunk 2]
```diff
@@ -46,27 +48,47 @@ public void close() throws Exception {
...
+    */
     @Override
     public void reset() {
         gyro.setFusedHeading(0);
 
     }
 
+    
+    /** 
+     * @return The rate in which we have traveled in degrees per second
```
Comment (msoucy) [inline core/src/main/java/com/chopshop166/chopshoplib/sensors/PigeonGyro.java:63]: Traveled or rotated?

File: core/src/main/java/com/chopshop166/chopshoplib/sensors/PigeonGyro.java [hunk 3]
```diff
@@ -46,27 +48,47 @@ public void close() throws Exception {
...
+     */
     @Override
     public double getRate() {
         final double[] xyz = new double[3];
         gyro.getRawGyro(xyz);
-        return xyz[2];
+        return inverted ? -xyz[2] : xyz[2];
```
Comment (msoucy) [inline core/src/main/java/com/chopshop166/chopshoplib/sensors/PigeonGyro.java:69]: Could you use `super.getRate()` instead of `getRawGyro()`?

**Review by msoucy — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/chopshop166/chopshoplib/sensors/PigeonGyro.java (hunk 2 above)
Comment (bot190) [reply core/src/main/java/com/chopshop166/chopshoplib/sensors/PigeonGyro.java:63]: Also this function gives the current rate, so the description should be in present tense.

**Pull request merged**